extensions/amp-fresh/0.1/amp-fresh.js

```diff
@@ -26,14 +29,53 @@ export class AmpFresh extends AMP.BaseElement {
 
   /** @override */
   isLayoutSupported(layout) {
-    return isLayoutSizeDefined(layout);
+    return isLayoutSizeDefined(layout) || layout == Layout.CONTAINER;
   }
 
   buildCallback() {
     /** @private @const {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.win, TAG);
 
     user().assert(this.isExperimentOn_, `Experiment ${TAG} disabled`);
+
+    this.ampFreshId_ = user().assert(this.element.getAttribute('id'),
+        'amp-fresh must have an id.');
+
+    /** @private @const {!AmpFreshManager} */
+    this.manager_ = installAmpFreshManager(this.win);
+
+    this.manager_.register(this.ampFreshId_, this);
+
+    this.hide();
+  }
+
+  /**
+   * @param {!Element} surrogateAmpFresh
+   */
+  update(surrogateAmpFresh) {
+    // Never reparent the surrogate to the current document's subtree
+    // as this will trigger custom element life cycles,
+    // importing it shouldn't trigger.
+    surrogateAmpFresh = this.win.document.adoptNode(surrogateAmpFresh);
+    this.mutateElement(() => {
+      this.element.textContent = '';
+      copyChildren(surrogateAmpFresh, this.element);
+      this.show();
+    });
+  }
+
+  /**
+   * Toggles the element to be visible
+   */
+  show() {
+    setStyle(this.element, 'visibility', 'visible');
```

Comment:
Actually, the reason why I'm bringing this up is b/c I don't think either `visibility:hidden` or `amp-hidden` are good ideas because it could be a significant type lapsed before this element is built. Thus `amp-hidden`/`visibility:hidden` would just be FOUC. Instead, maybe we should just rely on `amp-active`?

From page author's point of view, this should still be easy enough:
```
.info {
  color: gray; /* start grayed out until fresh information is received */
}
.amp-active .info {
  color: black; /* now it's fresh */
}
```

WDYT?

Also, `amp-active` doesn't communicate semantically what's happening, let's think of other names as well.